### ISSUE TITLE
Regarding Actor update

### ISSUE THREAD

**Issue by Erfi** [arnavkj1995/SFM#1]
Hello,

While trying to implement the SFM algorithm, I have come across a few questions regarding the actor update:

1) The pseudocode on the [website](https://arnavkj1995.github.io/SFM/) seems to be different than the one on your [paper](https://arxiv.org/pdf/2411.07007). The one on the website seems to be based on Feature Matching, and the one in the paper is taking an Actor-Critic approach of maximizing the Q while Q is represented as the product of SF and witness. 

2) Looking at the code (e.g. `sfm_td3.py`), I see the following:
https://github.com/arnavkj1995/SFM/blob/acd1979113d0a9d7aad3f265d00ae4bd5cf9295a/agents/sfm_td3.py#L319-L320
My understanding is that the actor is trying to match the SF of the expert to the SF of the policy (instead of maximizing the Q, represented by the product of SF of the policy and witness). In that case:
- Why use once `Pi_SF` and once `Pi_SF_ema`? 
- Where did the `(1.0 - gamma)` terms come from?

I will appreciate your clarification :)

**Comment by arnavkj1995** [comment]
Hey Erfan, 

Thanks for your interest in our work. 
1. For the website, I need to change the pseudocode to the version in the camera-ready for ICLR. The codebase is aligned with the paper. 
2. For the loss-- We maintain an EMA for the expected SF of the policy to ensure that the witness or reward function changes slowly for stable training. Is it not required to propagate gradients through the `Pi_SF_ema`, so this computation can be moved after line 320 or detached, and the code should work fine too. 
3. I used the `1-gamma` term to scale down the loss function because the loss values are quite large in magnitude. 

Do let us know if there are any further questions.

**Comment by Erfi** [comment]
Hi Arnav,

Thank you for your response! These made things quite clear.
I have implemented the actor loss also using Actor-Critic style (pytorch):
```
witness = expert_psi_ema_debiased - self.pi_psi_ema
actor_loss = -torch.matmul(pi_psi_current, witness)
actor_loss = actor_loss.mean() 
```
Seems to give more or less similar performance. Do you have any comments/intuition about the similarities or differences between these two ways of calculating Actor loss, or if one should be preferred over the other? 

Thanks again, 
Erfan

**Comment by arnavkj1995** [comment]
Hi Erfan,

They should perform similarly, as there should be no gradients through the EMA term in the loss. Your implementation looks correct and should work. 

Let me know if you end up releasing the Pytorch version.
